### PR TITLE
chore: Consolidate CodeMirror code files [WEB-1306]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

We want CodeEditor in UI Kit to be the one and only code viewing and editing component. This uses `ReactCodeMirror` directly in UI Kit's `CodeEditor`, removing need for a wrapper component.

Notes:
- Markdown continues to exist as a separate component because it includes both viewing and editing

## Test Plan

Files syntax highlighted in experiment:
- In the Code tab, switch between read-only YAML and Python files
- In the Notes tab, click the pencil icon to open a Markdown editor. Confirm you can type markdown and save.

Confirm editors look OK in mini height modals:
- In the top right of the experiment page, click Fork. In the modal, click Show Full Config to open the YAML editor
- In this fork experiment modal, in the YAML, change (likely multi-line) description to a single line "description: my test 12345" then click "Fork". The new experiment has your set description.
- In the left sidebar, hover over Workspaces. Click the "+" icon, then switch on "Configure Checkpoint Storage" to open a YAML editor

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
Netlify preview was cancelled?

**Comment by mapmeld** [comment]
I must have pushed / force-pushed a few times while the first Netlify preview was being created. I rebased to see if this can re-trigger Netlify

File: webui/react/src/components/ExperimentCreateModal.tsx [hunk 1]
```diff
@@ -365,10 +366,10 @@ const ExperimentCreateModalComponent = ({
         )}
         {modalState.isAdvancedMode && (
           <React.Suspense fallback={<Spinner tip="Loading text editor..." />}>
-            <CodeMirrorEditor
+            <CodeEditor
+              files={[{ content: Loaded(modalState.configString), key: 'config.yaml' }]}
               height="40vh"
-              syntax="yaml"
-              value={modalState.configString}
+              readonly={false}
```
[the remark this answers is not in the document]
Comment (mapmeld) [reply webui/react/src/components/ExperimentCreateModal.tsx:372]: :+1:

File: webui/react/src/components/kit/CodeEditor.tsx [hunk 2]
```diff
@@ -250,11 +283,14 @@ const CodeEditor: React.FC<Props> = ({ files, onSelectFile, readonly, selectedFi
   } else if (activeFile) {
     fileContent =
       editorMode === 'codemirror' ? (
-        <CodeMirrorEditor
-          height="100%"
+        <ReactCodeMirror
+          basicSetup={getSyntaxHighlight() === 'markdown' ? MARKDOWN_CONFIG : undefined}
+          extensions={[langs[getSyntaxHighlight()]()]}
```
[the remark this answers is not in the document]
Comment (mapmeld) [reply webui/react/src/components/kit/CodeEditor.tsx]: ok I think the latest commit fixes this. my proxy is being a bit weird, will double-check then merge

**Pull request merged**